test/utils/validateApiVersion.test.js

```diff
@@ -0,0 +1,29 @@
+import { validateApiVersion, isVersionDateInTheFuture } from '../../src/utils/validateApiVersion';
+
+describe('Validate the provided API version', () => {
+  it('should allow the provided API version', () => {
+    expect(validateApiVersion('2018-05-01')).toBe(true);
+  });
+
+  it('should throw an error for an invalid format', () => {
+    expect(() => {
+      validateApiVersion('01-05-20188');
+    }).toThrowError(
+      'The provided API version is not in the right format. Please provide the API version in the yyyy-mm-dd format.',
+    );
+  });
+
+  it('should throw an error because the date is in the future', () => {
+    expect(() => {
+      validateApiVersion('2030-01-01');
+    }).toThrowError('The provided API version date is in the future. Please provide an API version from the past.');
+  });
+
+  it('should recognize that the version date is in the past', () => {
+    expect(isVersionDateInTheFuture('2018-09-01')).toBe(false);
+  });
+
+  it('should recognize that the version date is in the future', () => {
+    expect(isVersionDateInTheFuture('2030-12-31')).toBe(true);
+  });
+});
```

Comment:
Maybe we should also add a test for an invalid date? 15 as month, 34 as day, not sure how easy that would be without lib.

Reply:
Hmm, actually I found out this about JS, I think it's a bit weird:
<img width="495" alt="screen shot 2018-12-11 at 10 35 12" src="https://user-images.githubusercontent.com/21224391/49791374-a7c89c80-fd30-11e8-8f48-4c51f6790e8c.png">

Reply:
If it isn't trivial to do I would keep it like this for now.